components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.hpp

```diff
@@ -303,10 +144,1035 @@ void mixing_ratio_to_cloud_mass(
+{
+  // Band 10 straddles the near-IR and visible, so divide contributions from band 10 between both broadband sums
+  // TODO: Hard-coding these band indices is really bad practice. If the bands ever were to change (like when
+  // the RRTMG bands were re-ordered for RRTMGP), we would be using the wrong bands for the IR and UV/VIS. This
+  // should be refactored to grab the correct bands by specifying appropriate wavenumber rather than index.
+  //sfc_flux_dir_nir(i) = sum(sw_bnd_flux_dir(i+1,kbot,1:9))   + 0.5 * sw_bnd_flux_dir(i+1,kbot,10);
+  //sfc_flux_dir_vis(i) = sum(sw_bnd_flux_dir(i+1,kbot,11:14)) + 0.5 * sw_bnd_flux_dir(i+1,kbot,10);
+  //sfc_flux_dif_nir(i) = sum(sw_bnd_flux_dif(i+1,kbot,1:9))   + 0.5 * sw_bnd_flux_dif(i+1,kbot,10);
+  //sfc_flux_dif_vis(i) = sum(sw_bnd_flux_dif(i+1,kbot,11:14)) + 0.5 * sw_bnd_flux_dif(i+1,kbot,10);
+
+  // Initialize sums over bands
+  Kokkos::deep_copy(sfc_flux_dir_nir, 0);
+  Kokkos::deep_copy(sfc_flux_dir_vis, 0);
+  Kokkos::deep_copy(sfc_flux_dif_nir, 0);
+  Kokkos::deep_copy(sfc_flux_dif_vis, 0);
+
+  // Threshold between visible and infrared is 0.7 micron, or 14286 cm^-1.
+  const RealT visible_wavenumber_threshold = 14286;
+  auto wavenumber_limits = k_dist_sw_k.get_band_lims_wavenumber();
+  Kokkos::parallel_for(ncol, KOKKOS_LAMBDA(const int icol) {
+    for (int ibnd = 0; ibnd < nswbands; ++ibnd) {
+      // Wavenumber is in the visible if it is above the visible wavenumber
+      // threshold, and in the infrared if it is below the threshold
+      const bool is_visible_wave1 = (wavenumber_limits(0, ibnd) > visible_wavenumber_threshold ? true : false);
+      const bool is_visible_wave2 = (wavenumber_limits(1, ibnd) > visible_wavenumber_threshold ? true : false);
+
+      if (is_visible_wave1 && is_visible_wave2) {
+        // Entire band is in the visible
+        sfc_flux_dir_vis(icol) += sw_bnd_flux_dir(icol,ktop,ibnd);
+        sfc_flux_dif_vis(icol) += sw_bnd_flux_dif(icol,ktop,ibnd);
+      }
+      else if (!is_visible_wave1 && !is_visible_wave2) {
+        // Entire band is in the longwave (near-infrared)
+        sfc_flux_dir_nir(icol) += sw_bnd_flux_dir(icol,ktop,ibnd);
+        sfc_flux_dif_nir(icol) += sw_bnd_flux_dif(icol,ktop,ibnd);
+      }
+      else {
+        // Band straddles the visible to near-infrared transition, so put half
+        // the flux in visible and half in near-infrared fluxes
+        sfc_flux_dir_vis(icol) += 0.5 * sw_bnd_flux_dir(icol,ktop,ibnd);
+        sfc_flux_dif_vis(icol) += 0.5 * sw_bnd_flux_dif(icol,ktop,ibnd);
+        sfc_flux_dir_nir(icol) += 0.5 * sw_bnd_flux_dir(icol,ktop,ibnd);
+        sfc_flux_dif_nir(icol) += 0.5 * sw_bnd_flux_dif(icol,ktop,ibnd);
+      }
+    }
+  });
+}
+
+/*
+ * Main driver code to run RRTMGP.
+ * The input logger is in charge of outputing info to
+ * screen and/or to file (or neither), depending on how it was set up.
+ */
+template <typename PlayT, typename TlayT, typename PlevT, typename TlevT,
```

Comment:
I guess you will probably answer this in the above comments I made, but again here we shouldn't have to assume that each input can have different types. The only types we should need to template on are those with different ranks, and there should only be maybe 4-5 different layouts I believe.